frontend/src/app/main/data/workspace/libraries.cljs

```diff
@@ -701,14 +714,22 @@
    (ptk/reify ::launch-component-sync
      ptk/WatchEvent
      (watch [_ state _]
-       (let [current-file-id (:current-file-id state)
+       (let [data            (get state :workspace-data)
+             component       (ctkl/get-component data component-id)
+             page-id         (:main-instance-page component)
+             root-id         (:main-instance-id component)
+             current-file-id (:current-file-id state)
              undo-id         (js/Symbol)]
-         (rx/of
-          (dwu/start-undo-transaction undo-id)
-          (sync-file current-file-id file-id :components component-id undo-group)
-          (when (not= current-file-id file-id)
-            (sync-file file-id file-id :components component-id undo-group))
-          (dwu/commit-undo-transaction undo-id)))))))
+          (rx/of
+           (dwu/start-undo-transaction undo-id)
+           (sync-file current-file-id file-id :components component-id undo-group)
+           (when (not= current-file-id file-id)
+             (sync-file file-id file-id :components component-id undo-group))
+           (dwu/commit-undo-transaction undo-id)
+           ;; Select the component root
+           (dwt/update-thumbnail current-file-id page-id root-id)
+           (when (not= current-file-id file-id)
+             (dwt/update-thumbnail file-id page-id root-id))))))))
```

Comment:
Don't understand this well.

I think it should be better to do the update-thumbnail in `watch-component-changes`. For each changed component, launch a sync, and then update the thumbnail.

The `launch-component-sync` function does one thing: launching a sync. If we add the thumbnail here, we are adding two responsabilities.